herbert_core/classes/data_loaders/@rundata/private/set_param_recursively.m

```diff
@@ -15,46 +15,62 @@
     if isstruct(varargin{1})
         obj=set_param_recursively(obj,varargin{1},varargin{2:end});
     else
-        obj=parse_arg(obj,varargin{:});        
+        obj=parse_arg(obj,varargin{:});
     end
 elseif isstruct(a_struct)
     field_names    = fieldnames(a_struct);
-    targ_fields    = fieldnames(obj);
-    field_values   = cell(numel(field_names),1);
-    lat_fields = oriented_lattice.lattice_fields;
-    if any(ismember(field_names,lat_fields))
-        change_sample = true;
-    else
-        change_sample = false;        
+    dir_fields = obj.saveableFields;
+    % check if some fields can be set up directly on the
+    [obj,field_names]=set_fields_subset(obj,a_struct,dir_fields,field_names);
+    if isempty(field_names)
+        return;
     end
-    for i=1:numel(field_names)
-        field_values{i} = a_struct.(field_names{i});
+    % check the fields, defined by lattice
+    lat = obj.lattice;
+    if isempty(lat)
+        lat = oriented_lattice();
     end
-    obj=set_fields_skip_special(obj,field_names,field_values,targ_fields);
-    if numel(varargin)>0
-        obj = set_param_recursively(obj,varargin{1},varargin{2:end});
+    lat_fields = lat.saveableFields();
```

Comment:
because this is what you may set up. To be fair, it would be nice to make better integration with saveable and throw half of this code, but I did not bothered at this stage. I think we should merge `IX_samp`, `oriented_lattice` and `IX_experiment` into something common, and prettify the final result, Not these sketches.